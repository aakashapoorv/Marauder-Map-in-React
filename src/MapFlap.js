import React, { PureComponent } from "react";
import PropTypes from "prop-types";

export default class MapFlap extends PureComponent {
  render() {
    const { number } = this.props;
    const cls = `map-flap flap---${number}`;
    return (
      <div className={cls}>
        <div className="map-flap__front" />
        <div className="map-flap__back" />
      </div>
    );
  }
}

MapFlap.propTypes = {
  number: PropTypes.number.isRequired
};
