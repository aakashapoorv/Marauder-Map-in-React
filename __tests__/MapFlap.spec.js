import React from "react";
import MapFlap from "../src/MapFlap";
import renderer from "react-test-renderer";
import Enzyme, { shallow, render, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

Enzyme.configure({ adapter: new Adapter() });

global.shallow = shallow;
global.render = render;
global.mount = mount;

describe("MapFlap tests", () => {
  let MapFlapElement;

  beforeEach(() => {
    MapFlapElement = renderer.create(<MapFlap number={1} />);
  });

  it("Should Render correctly", () => {
    let tree = MapFlapElement.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
