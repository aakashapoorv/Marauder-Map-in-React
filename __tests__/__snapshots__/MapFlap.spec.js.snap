// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapFlap tests Should Render correctly 1`] = `
<div
  className="map-flap flap---1"
>
  <div
    className="map-flap__front"
  />
  <div
    className="map-flap__back"
  />
</div>
`;
